core/libdeno/inspector.h

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+#ifndef INSPECTOR_H_
+#define INSPECTOR_H_
+
+#include <memory>
+#include <string>
+
+#include "third_party/v8/include/libplatform/libplatform.h"
+#include "third_party/v8/include/v8-inspector.h"
+#include "third_party/v8/include/v8.h"
+#include "third_party/v8/src/base/macros.h"
+
+#include "internal.h"
+
+namespace deno {
+class DenoIsolate;
+}
+
+namespace v8 {
+
+enum { kInspectorClientIndex };
+
+class InspectorFrontend final : public v8_inspector::V8Inspector::Channel {
+ public:
+  explicit InspectorFrontend(Local<Context> context);
+  ~InspectorFrontend() override = default;
+
+  deno::DenoIsolate* deno_;
+
+ private:
+  void sendResponse(
+      int callId, std::unique_ptr<v8_inspector::StringBuffer> message) override;
+  void sendNotification(
+      std::unique_ptr<v8_inspector::StringBuffer> message) override;
+
+  void flushProtocolNotifications() override {}
+
+  void Send(const v8_inspector::StringView& string);
+
+  Isolate* isolate_;
+  Global<Context> context_;
+};
+
+class InspectorClient : public v8_inspector::V8InspectorClient {
+ public:
+  InspectorClient(Local<Context> context, deno::DenoIsolate* deno_);
+  ~InspectorClient() override = default;
+
+  void runMessageLoopOnPause(int context_group_id) override;
```

Comment:
For DevTools and tools to work properly some other methods are useful:
- ensureDefaultContextInGroup - method should return default `v8::Context`,
- currentTimeMS - method is used by different profilers.

Reply:
.. and definitely `runIfWaitingForDebugger` - this method is called when inspector backend is processing `Runtime.runIfWaitinForDebugger` protocol command.